File: LmsApplication.CourseBoardModule.Data/Models/GradeModel.cs
using LmsApplication.CourseBoardModule.Data.Entities;

namespace LmsApplication.CourseBoardModule.Data.Models;

public class GradeModel
{
    public required GradesTableRowDefinitionModel Row { get; set; }
    
    public required GradesTableRowValueModel? Value { get; set; }
}

public class RowGradesModel
{
    public required GradesTableRowDefinitionModel Row { get; set; }
    
    public required List<UserGradesTableRowValueModel> Values { get; set; }
}

public class UserGradesModel
{
    public required List<GradeModel> Grades { get; set; }
    
    public required FinalGradeModel? FinalGrade { get; set; }
}

File: LmsApplication/LmsApplication.Frontend/src/app/types/course-board/grade-model.ts
import { UserModel } from '../users/user-model';
import { GradesTableRowModel } from './grades-table-row-model';
import { GradesTableRowValueModel } from './user-grades-model';

export interface GradeModel {
  row: GradesTableRowModel;
  value: GradesTableRowValueModel | null;
}

export interface UserGradeModel {
  grades: GradeModel[];
  finalGrade: FinalGradeModel;
}

export interface FinalGradeModel {
  id: string;
  value: number;
  teacher: UserModel;
}
